tests/apply/partial.c

```diff
@@ -0,0 +1,84 @@
+#include "clar_libgit2.h"
+#include "git2/sys/repository.h"
+
+#include "patch.h"
+#include "repository.h"
+#include "buf_text.h"
+
+#include "../patch/patch_common.h"
+
+static git_repository *repo = NULL;
+
+void test_apply_partial__initialize(void)
+{
+	repo = cl_git_sandbox_init("renames");
+}
+
+void test_apply_partial__cleanup(void)
+{
+	cl_git_sandbox_cleanup();
+}
+
+static int skip_addition(
+	const git_diff_delta *delta,
+	const git_diff_hunk *hunk,
+	void *payload)
+{
+	GIT_UNUSED(delta);
+	GIT_UNUSED(payload);
+
+	return (hunk->new_lines > hunk->old_lines) ? 1 : 0;
+}
```

Comment:
Would be nice to also have a test which aborts adding hunks by returning `-1`. Additional kudos if aborting mid-way, that is after adding a few other hunks.